### DISCUSSION TITLE
What about React Server Components?

### DISCUSSION THREAD

**Discussion by popuguytheparrot** [TanStack/router#802] [General]
What about React Server Components?
Is this possible?
What problems?

**Comment by CanRau** [comment]
From [Discord](https://discord.com/channels/719702312431386674/1238170697650405547/1277086366999642174) by Tanner Linsley

> RSCs will be available soon. Since they have to happen on the server, you’ll simply return jsx/elements from a createServerFn

**Reply by osdiab** [reply]
Any updates on this? Been a while since then

**Comment by aretrace** [comment]
@popuguytheparrot Example repo: https://github.com/TanStack/router/tree/main/e2e/start/basic-rsc

**Reply by CanRau** [reply]
Uuuh fantastic thanks for sharing 🙏🏼

**Reply by osdiab** [reply]
FYI it appears this is not supported regardless of there being an example: https://github.com/TanStack/router/issues/4382

Relatedly it looks like Vite is making progress on RSC support, not sure if that has bearing on this: https://github.com/vitejs/vite-plugin-react/issues/531

**Reply by aretrace** [reply]
Looks like @osdiab is right, @schiller-manuel mentions that RSC is currently not supported 😢

**Comment by vramanalj** [comment]
Any update on this ?

**Comment by lovrozagar** [comment]
Any updates? Thanks

**Reply by schiller-manuel** [reply]
we are working on this. ongoing effort. we will let you know when we have something to play with

**Reply by lovrozagar** [reply]
Thank you

**Reply by Bluzzi** [reply]
@schiller-manuel will Server Components be optional, or do you plan to enforce their use? For example, will it be possible to enable or disable them via configuration for users who don’t want to use them?

**Reply by schiller-manuel** [reply]
@Bluzzi server components will be fully opt-in. if you dont want to use it, you dont do anything, if you want to use them, then you add an additional vite plugin and specify some config values..

**Comment by Neonsy** [comment]
I'm the next guy asking for an update 😂 

I love the framework, but I need RSC for what I'm planning

It's the only thing missing 🙏

I know it's important to do it right, I'm just wondering if a rough ETA exists 🙃
